src/zenml/zen_server/routers/flavors_endpoints.py

```diff
@@ -102,60 +113,50 @@ def get_flavor(
 @handle_exceptions
 def create_flavor(
     flavor: FlavorRequest,
-    auth_context: AuthContext = Security(
-        authorize, scopes=[PermissionType.WRITE]
-    ),
+    _: AuthContext = Security(authorize),
 ) -> FlavorResponse:
     """Creates a stack component flavor.
 
     Args:
         flavor: Stack component flavor to register.
-        auth_context: Authentication context.
 
     Returns:
         The created stack component flavor.
-
-    Raises:
-        IllegalOperationError: If the workspace or user specified in the stack
-            component flavor does not match the current workspace or authenticated
-            user.
     """
-    if flavor.user != auth_context.user.id:
-        raise IllegalOperationError(
-            "Creating flavors for a user other than yourself "
-            "is not supported."
-        )
-
-    created_flavor = zen_store().create_flavor(
-        flavor=flavor,
+    return verify_permissions_and_create_entity(
+        request_model=flavor,
+        resource_type=ResourceType.FLAVOR,
+        create_method=zen_store().create_flavor,
     )
-    return created_flavor
 
 
 @router.put(
-    "/{team_id}",
+    "/{flavor_id}",
     response_model=FlavorResponse,
     responses={401: error_response, 409: error_response, 422: error_response},
 )
 @handle_exceptions
 def update_flavor(
     flavor_id: UUID,
     flavor_update: FlavorUpdate,
-    _: AuthContext = Security(authorize, scopes=[PermissionType.WRITE]),
+    _: AuthContext = Security(authorize),
 ) -> FlavorResponse:
     """Updates a flavor.
 
     # noqa: DAR401
 
     Args:
-        flavor_id: ID of the team to update.
-        flavor_update: Team update.
+        flavor_id: ID of the flavor to update.
+        flavor_update: Flavor update.
 
     Returns:
         The updated flavor.
     """
-    return zen_store().update_flavor(
-        flavor_id=flavor_id, flavor_update=flavor_update
+    return verify_permissions_and_update_entity(
```

Comment:
Same thing here: the flavor model doesn't inherit from `UserScopedResponseModel` so the owner won't be recognized by these utility functions.

Reply:
Flavors are now user-scoped as you suggested, nice catch!